components/channel/include/channel.h

```diff
@@ -57,8 +57,10 @@ struct channel {
 };
 
 typedef struct channel Channel;
-typedef struct channel Channel_in;
-typedef struct channel Channel_out;
+typedef struct channel Channel_in __attribute__((deprecated));
+typedef struct channel Channel_out __attribute__((deprecated));
```

Comment:
Why use deprecated here? Nothing containing the channel structure has been merged so far. Why not remove it?